UCX compilation failure on Fedora 36

### Describe the bug
UCX compilation fails on Fedora 36 (beta) with gcc `gcc (GCC) 12.0.1 20220413 (Red Hat 12.0.1-0)`

```
CXX      ucm/gtest-malloc_hook.o
In member function ‘bool test_thread::is_ptr_in_range(void*, size_t, const std::vector<std::pair<void*, void*> >&)’,
    inlined from ‘void test_thread::test()’ at ../../../test/gtest/ucm/malloc_hook.cc:417:5:
../../../test/gtest/ucm/malloc_hook.cc:306:38: error: pointer may be used after ‘void free(void*)’ [-Werror=use-after-free]
  306 |             if ((ptr >= iter->first) && ((char*)ptr < iter->second)) {
      |                 ~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../../test/gtest/ucm/malloc_hook.cc: In member function ‘void test_thread::test()’:
../../../test/gtest/ucm/malloc_hook.cc:414:9: note: call to ‘void free(void*)’ here
  414 |     free(ptr);
      |     ~~~~^~~~~
In member function ‘bool test_thread::is_ptr_in_range(void*, size_t, const std::vector<std::pair<void*, void*> >&)’,
    inlined from ‘void test_thread::test()’ at ../../../test/gtest/ucm/malloc_hook.cc:417:5:
../../../test/gtest/ucm/malloc_hook.cc:306:38: error: pointer may be used after ‘void free(void*)’ [-Werror=use-after-free]
  306 |             if ((ptr >= iter->first) && ((char*)ptr < iter->second)) {
      |                 ~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../../test/gtest/ucm/malloc_hook.cc: In member function ‘void test_thread::test()’:
../../../test/gtest/ucm/malloc_hook.cc:414:9: note: call to ‘void free(void*)’ here
  414 |     free(ptr);
      |     ~~~~^~~~~
In member function ‘bool test_thread::is_ptr_in_range(void*, size_t, const std::vector<std::pair<void*, void*> >&)’,
    inlined from ‘void test_thread::test()’ at ../../../test/gtest/ucm/malloc_hook.cc:452:5:
../../../test/gtest/ucm/malloc_hook.cc:306:38: error: pointer may be used after ‘void operator delete(void*)’ [-Werror=use-after-free]
  306 |             if ((ptr >= iter->first) && ((char*)ptr < iter->second)) {
      |                 ~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/c++/12/aarch64-redhat-linux/bits/c++allocator.h:33,
                 from /usr/include/c++/12/bits/allocator.h:46,
                 from /usr/include/c++/12/memory:64,
                 from ../../../test/gtest/common/googletest/gtest.h:57,
                 from ../../../test/gtest/common/test_helpers.h:11,
                 from ../../../test/gtest/common/test.h:23,
                 from ../../../test/gtest/ucm/malloc_hook.cc:13:
In member function ‘void std::__new_allocator<_Tp>::deallocate(_Tp*, size_type) [with _Tp = char]’,
    inlined from ‘static void std::allocator_traits<std::allocator<_Tp1> >::deallocate(allocator_type&, pointer, size_type) [with _Tp = char]’ at /usr/include/c++/12/bits/alloc_traits.h:496:23,
    inlined from ‘void std::_Vector_base<_Tp, _Alloc>::_M_deallocate(pointer, std::size_t) [with _Tp = char; _Alloc = std::allocator<char>]’ at /usr/include/c++/12/bits/stl_vector.h:386:19,
    inlined from ‘std::_Vector_base<_Tp, _Alloc>::~_Vector_base() [with _Tp = char; _Alloc = std::allocator<char>]’ at /usr/include/c++/12/bits/stl_vector.h:365:15,
    inlined from ‘std::vector<_Tp, _Alloc>::~vector() [with _Tp = char; _Alloc = std::allocator<char>]’ at /usr/include/c++/12/bits/stl_vector.h:732:7,
    inlined from ‘void test_thread::test()’ at ../../../test/gtest/ucm/malloc_hook.cc:449:5:
/usr/include/c++/12/bits/new_allocator.h:158:33: note: call to ‘void operator delete(void*)’ here
  158 |         _GLIBCXX_OPERATOR_DELETE(_GLIBCXX_SIZED_DEALLOC(__p, __n));
      |                                 ^
In member function ‘bool test_thread::is_ptr_in_range(void*, size_t, const std::vector<std::pair<void*, void*> >&)’,
    inlined from ‘void test_thread::test()’ at ../../../test/gtest/ucm/malloc_hook.cc:452:5:
../../../test/gtest/ucm/malloc_hook.cc:306:38: error: pointer may be used after ‘void operator delete(void*)’ [-Werror=use-after-free]
  306 |             if ((ptr >= iter->first) && ((char*)ptr < iter->second)) {
      |                 ~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In member function ‘void std::__new_allocator<_Tp>::deallocate(_Tp*, size_type) [with _Tp = char]’,
    inlined from ‘static void std::allocator_traits<std::allocator<_Tp1> >::deallocate(allocator_type&, pointer, size_type) [with _Tp = char]’ at /usr/include/c++/12/bits/alloc_traits.h:496:23,
    inlined from ‘void std::_Vector_base<_Tp, _Alloc>::_M_deallocate(pointer, std::size_t) [with _Tp = char; _Alloc = std::allocator<char>]’ at /usr/include/c++/12/bits/stl_vector.h:386:19,
    inlined from ‘std::_Vector_base<_Tp, _Alloc>::~_Vector_base() [with _Tp = char; _Alloc = std::allocator<char>]’ at /usr/include/c++/12/bits/stl_vector.h:365:15,
    inlined from ‘std::vector<_Tp, _Alloc>::~vector() [with _Tp = char; _Alloc = std::allocator<char>]’ at /usr/include/c++/12/bits/stl_vector.h:732:7,
    inlined from ‘void test_thread::test()’ at ../../../test/gtest/ucm/malloc_hook.cc:449:5:
/usr/include/c++/12/bits/new_allocator.h:158:33: note: call to ‘void operator delete(void*)’ here
  158 |         _GLIBCXX_OPERATOR_DELETE(_GLIBCXX_SIZED_DEALLOC(__p, __n));
      |                                 ^
```

--- a/test/gtest/ucm/malloc_hook.cc
+++ b/test/gtest/ucm/malloc_hook.cc
@@ -302,8 +302,12 @@ private:
     typedef std::pair<void*, void*> range;
 
     bool is_ptr_in_range(void *ptr, size_t size, const std::vector<range> &ranges) {
-        for (std::vector<range>::const_iterator iter = ranges.begin(); iter != ranges.end(); ++iter) {
-            if ((ptr >= iter->first) && ((char*)ptr < iter->second)) {
+        uintptr_t p = (uintptr_t)ptr;
+        for (std::vector<range>::const_iterator iter = ranges.begin();
+             iter != ranges.end(); ++iter) {
+            uintptr_t begin = (uintptr_t)iter->first;
+            uintptr_t end   = (uintptr_t)iter->second;
+            if ((p >= begin) && (p < end)) {
                 return true;
             }
         }

